Kudu.Core/Deployment/Generator/ExternalCommandBuilder.cs

```diff
@@ -204,5 +212,23 @@ public IList<string> GetPostBuildActionScripts()
 
             return scriptFilesGroupedAndSorted;
         }
+
+        private string GetLinuxPostBuildFilepath(FileInfo fi)
+        {
+            if (string.Equals(".sh", fi.Extension, StringComparison.OrdinalIgnoreCase))
```

Comment:
Does this file get Execute permissions latest in the deployment flow ?

Reply:
No, but from my understanding, I think a read permission is only needed to execute the script from a specific shell.

Reply:
We run it as ./script.sh, it would need an execute permission. Look at PermissionHelper.Chmod("ugo+x", scriptPath, Environment, DeploymentSettings, buildLogger);

Reply:
Thanks for pointing this out. The issue is now addressed.

Reply:
Looks good! I'll merge it with dev